feature-toggles/feature-toggles-impl/src/main/java/com/duckduckgo/feature/toggles/impl/MetricsPixelStore.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.feature.toggles.impl
+
+import android.content.SharedPreferences
+import androidx.annotation.WorkerThread
+import androidx.core.content.edit
+import com.duckduckgo.appbuildconfig.api.AppBuildConfig
+import com.duckduckgo.appbuildconfig.api.isInternalBuild
+import com.duckduckgo.common.utils.checkMainThread
+import com.duckduckgo.data.store.api.SharedPreferencesProvider
+import com.duckduckgo.di.scopes.AppScope
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface MetricsPixelStore {
+    /**
+     * @return true if the pixel with the given tag can be fired or false otherwise
+     */
+    @WorkerThread
+    fun wasPixelFired(tag: String): Boolean
+
+    /**
+     * Stores the tag [String] passed as parameter
+     */
+    @WorkerThread
+    fun storePixelTag(tag: String)
```

Comment:
No strong opinion about this because we have both patterns in the code base, I have used both of them and this is in an impl module.
For this cases I nowadays fancy more `suspend fun` for getters that clearly access disk IO and define the IO dispatcher internally.
And normal `fun` for setters that internally do `launch` in `dispatchers.io`